woocommerce/woocommerce
# Product Block Editor: improve the process when creating a new attribute

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR improves the process when creating a new attribute from the attributes modal.It gets rid of the weird `-99` attribute ID.





Feedback: https://github.com/woocommerce/woocommerce/pull/47093#discussion_r1592573552
Closes https://github.com/woocommerce/woocommerce/issues/47263



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

This change should not alter the app's functionality, so it comes down to verifying the ability to create new attributes, either globally or locally.

### Test 

1. Enable the new Product Editor
2. Go to the `Variations` section
3. Click on the `Add options` button
4. Type a new attribute
5. Confirm it creates the new attribute when clicking on the `Create "..."` option or pressing the `Enter` key. You can check the async requests:

<img width="807" alt="image" src="https://github.com/woocommerce/woocommerce/assets/77539/0533199d-54c5-4acb-b841-c1a119622889">

7. Go to the `Organization` section
8. Click on the `Add new` button 
9. Type into the input to create a new attribute
10. Confirm it does not add it as global, confirming the client doesn't perform the request 




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
